crates/vim/src/surrounds.rs

```diff
@@ -0,0 +1,589 @@
+use crate::{motion::Motion, object::Object, state::Mode, Vim};
+use editor::{scroll::Autoscroll, Bias};
+use gpui::WindowContext;
+use language::BracketPair;
+use serde::Deserialize;
+use std::ops::Deref;
+use std::sync::Arc;
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize)]
+pub enum SurroundsType {
+    Motion(Motion),
+    Object(Object),
+}
+
+pub fn add_surrounds(text: Arc<str>, target: SurroundsType, cx: &mut WindowContext) {
+    Vim::update(cx, |vim, cx| {
+        vim.stop_recording();
+        vim.update_active_editor(cx, |_, editor, cx| {
+            let text_layout_details = editor.text_layout_details(cx);
+            editor.transact(cx, |editor, cx| {
+                editor.set_clip_at_line_ends(false, cx);
+                editor.change_selections(Some(Autoscroll::fit()), cx, |s| {
```

Comment:
Instead of changing the selection and then replacing the whole selection I think we should use `object.range` and `motion.move_point` to get the start and end locations and push one edit for the start and one for the end.

Reply:
(same for delete_surrounds and change_surrounds). This will reduce editing conflicts and cursor confusion if a collaborator's cursor is in the range.